Mimic ChatGPT Like Conversation History

techwithanirudh opened the pull request:
## Summary

This P.R adds support for infinite scrolling and grouping conversations by date, thereby improving the organization of conversations and enhancing user experience. In this P.R, conversations are sorted based on their date and grouped by the current date, with group text as a heading. The addition of infinite scrolling makes the conversation page more user-friendly and reduces the need for pagination.

## Context and Motivation

Grouping conversations by date is a valuable feature that enhances the organization and accessibility of conversations. By grouping conversations based on the date, users can quickly and easily locate specific conversations from a certain time period. The addition of infinite scrolling further improves the user experience of our application by removing the need for pagination.

## Testing

We tested this feature extensively to ensure that conversations are sorted and grouped correctly by their date and that infinite scrolling works smoothly.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective, or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

techwithanirudh committed:
in

techwithanirudh committed:
Merge branch 'danny-avila:main' into history

techwithanirudh committed:
Fix bugs

techwithanirudh committed:
Merge branch 'danny-avila:main' into history

techwithanirudh committed:
Final bug fix

techwithanirudh commented:
@ClaraLeigh

techwithanirudh committed:
Refactor the Nav component to enable paging of conversations in the chat application.

techwithanirudh committed:
Refactor conversations component rendering and navigation bar display

- Simplify the grouping of conversations in the component to use an ordered list instead of separate sections.
- Modify the heading level of the group titles for better accessibility.
- Remove unused variables in the navigation bar component and adjust its layout to match the revamped conversations component.

techwithanirudh committed:
Refactor conversations pagination logic.

This commit modifies the pagination logic and adds the "pages" variable to the "getConversationsQuery.data" object. Sets the "hasMorePages" state to false when the last page is reached instead of when there are no conversations left.

techwithanirudh committed:
Fixes

techwithanirudh committed:
fix

ClaraLeigh commented on package.json:
```diff
@@ -49,5 +49,8 @@
     "prettier-plugin-tailwindcss": "^0.2.2"
+  },
+  "dependencies": {
+    "react-infinite-scroll-component": "^6.1.0"
```
This is for the front end yeah? This package should be added to the client/package.json then. It'll still be installed and added to the package-lock but it'll be specific to the client workspace (that's why you don't see the other deps in this file)

ClaraLeigh commented:
Also remember to fill out the checklist on this PR

techwithanirudh committed:
Improve infinite scroll performance
- Update package.json dependencies to reference a manually fixed build of 'react-infinite-scroll-component' Git repository
- Increase Nav scroll threshold to 0.9
- Remove 'react-infinite-scroll-component' dependency from package-lock.json and package.json, which are no longer used.

techwithanirudh committed:
Add React infinite scroll component

This commit adds the React infinite scroll component to the list of dependencies in package.json and modifies the loader component in the Nav component to show a spinner animation; the end message is also set to null. The package-lock.json file is also updated to include the new version of the React infinite scroll component.

techwithanirudh committed:
Use git upstream

techwithanirudh committed:
Add package-lock.json

techwithanirudh committed:
Revert "Add package-lock.json"

This reverts commit 076d40ff54c8b312d398c12f84f30606438a0ed2.

techwithanirudh committed:
Use loading instead

techwithanirudh committed:
Revert "Use loading instead"

This reverts commit f4a42f1a08cbc4276deac6becbd27635e880c5c4.

techwithanirudh committed:
Revert "Revert "Add package-lock.json""

This reverts commit cd76758664107c49e8bff9bbf86700902c9b8d4e.

techwithanirudh committed:
Update git npmjs packages.

techwithanirudh committed:
Add yarn

techwithanirudh committed:
Migration from yarn to npm

ClaraLeigh commented on client/package.json:
```diff
@@ -60,6 +61,7 @@
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-hook-form": "^7.43.9",
+    "react-infinite-scroll-component": "github:techwithanirudh/react-infinite-scroll-component",
```
I'm not sure that this will meet the stability requirements for this project, unless you're still working on this? might be best to change the PR back to a draft then

techwithanirudh replied:
Ok.

techwithanirudh committed:
Merge remote-tracking branch 'upstream/main' into history
